static/app/components/events/autofix/autofixDrawer.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import {AutofixDataFixture} from 'sentry-fixture/autofixData';
+import {AutofixStepFixture} from 'sentry-fixture/autofixStep';
+import {EventFixture} from 'sentry-fixture/event';
+import {GroupFixture} from 'sentry-fixture/group';
+import {ProjectFixture} from 'sentry-fixture/project';
+
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {AutofixDrawer} from 'sentry/components/events/autofix/autofixDrawer';
+import {t} from 'sentry/locale';
+
+// Mock the hook used in the component
+jest.mock('sentry/components/events/autofix/useAutofix', () => ({
+  useAiAutofix: jest.fn(),
+}));
+
+// Mock the route analytics hook
+jest.mock('sentry/utils/routeAnalytics/useRouteAnalyticsParams', () => jest.fn());
+
+describe('AutofixDrawer', () => {
+  const mockEvent = EventFixture();
+  const mockGroup = GroupFixture();
+  const mockProject = ProjectFixture();
+
+  const mockAutofixData = AutofixDataFixture({steps: [AutofixStepFixture()]});
+
+  const mockTriggerAutofix = jest.fn();
+  const mockReset = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    require('sentry/components/events/autofix/useAutofix').useAiAutofix.mockReturnValue({
```

Comment:
Instead of mocking the hook completely, it's better to add mocks for the http responses instead. This tests the logic better (because it tests the integration between the component and the hook) and is also a bit less prone to breaking from refactors.

You can do something like:

```
MockApiClient.addMockResponse({
  url: `/issues/${group.id}/autofix/`,
  body: {autofix: autofixData},
});
```

When mocking API responses, you'll need to use async assertions to wait for changes to take place. So you'll want the first assertion to be something `screen.findByText('...')`

Reply:
Here, `mockReset` isn't an http request, though. It's just some callback in the hook. I agree `triggerAutofix` is an http request, but all I want to check is that it's sent; nothing about handling any responses.

How would I adjust the test in this case?

Reply:
Instead of asserting that `mockReset()` is called, you want to test for the UI to have updated correctly. For component tests you want to act like you're a user. The user doesn't care that you called a function, they care that Autofix is showing the correct thing and that the reset button let's them trigger a new run

Reply:
Makes sense, I updated the tests with this approach in the next commit